client/rest-high-level/src/test/java/org/elasticsearch/client/CrudIT.java

```diff
@@ -65,6 +64,11 @@
     public void testDelete() throws IOException {
         {
             // Testing non existing document
+            final XContentType xContentType = randomFrom(XContentType.values());
```

Comment:
I think you still need to use the low level client for that, using `client().performRequest()`. That said, why not change the order of the test blocks around and move the `// Testing deletion` block to be first. It will create the index.